kong/core/balancer.lua

```diff
@@ -470,7 +517,7 @@ local function get_balancer(target, no_create)
       return nil, "balancer not found"
     else
       log(ERR, "balancer not found for ", upstream.name, ", will create it")
-      return create_balancer(upstream)
+      return create_balancer(upstream), upstream
```

Comment:
dammit, seems like this already was an issue?

Reply:
it wasn't -- I mean, it was with regard to the function signature but it didn't happen because the `get_balancer` -> `create_balancer` code path was never hit when all balancers were created at startup. Now that it is async and `get_balancer` can run before `init` finishes running, then this code path can happen.

Reply:
My one concern here is that this can be confusing for users of get_balancer and increase cognitive load: `create_balancer` can return 1 or 2 values, making `get_balancer` return 1, 2, or 3 values. Usually, we deal with this by catching errors like:
```lua
local balancer, err = create_balancer(upstream)
if not balancer then
  return nil, err
end

return balancer, upstream -- still somewhat annoying to me, I prefer `balancer, nil, upstream` but up to you!
```